### ISSUE TITLE
Guide for Azure AD SAMLv2

### ISSUE THREAD

**Issue by smcoll** [FusionAuth/fusionauth-site#29]
i would have appreciated some more guidance configuring a SAML v2 Identity Provider for Azure AD (not ADFS).  Here's what i've done so far, to check my assumptions, and in case anyone else is trying to do the same:

Ref: https://docs.microsoft.com/en-us/azure/active-directory/manage-apps/configure-single-sign-on-non-gallery-applications

- installed FusionAuth (1.15.2) and made it available over DNS (i used ngrok to connect localhost:9011 to a reserved subdomain, for the time being)
- configured an Azure Enterprise ("non-gallery") application with SSO support via SAML v2.  The "Single sign-on" page looks something like this:
   1) Basic SAML Configuration
      - _Identifier (Entity ID)_: used FusionAuth issuer (see below)
      - _Reply URL_: used FusionAuth SAML callback/ACS URL (see below)
   2) User Attributes & Claims
      - not sure yet how to best configure assigning `user.email` and `user.principalname` to SAML claim names
   3) SAML Signing Certificate
      - downloaded base64 key, added it in FusionAuth Key Master
   4) Set up {appname}
      - noted Login URL
   5) Test single sign-on with {appname}
      - this lands me at `/samlv2/acs` with an error looking like:
        ```json
        {
          "error" : "invalid_request",
          "error_description" : "The request is missing a required parameter: redirect_uri",
          "error_reason" : "missing_redirect_uri"
        }
        ```
        That looks like an OAuth error, so i'm assuming this means the Azure test doesn't follow up with an adequate OAuth flow?
- configured a FusionAuth Identity provider of SAML v2 type
  - _IdP endpoint_: used Enterprise app Login URL (from step iv. above)
  - _Verification key_: used key created in FusionAuth Key Manager from step iv. above)
  - noted ACS and Issuer for Enterprise app config (for step i. above)

i _do_ have this working for the FusionAuth app (it creates the user, but auto-registration isn't supported) - after i create the registration manually, i can see that it works. 

Is SAML v2 Logout URL supported in FusionAuth's SAML configuration?

How should i handle mapping user attibutes and claims?  What claims are FusionAuth expecting?

Am i on track with my assumptions about the response to the Enterprise App SSO test's response?

**Comment by mooreds** [comment]
@smcoll sorry for the delay. Did you get this working?

**Comment by smcoll** [comment]
i did.  i think FA has since changed how the claim reconciliation lambda works, but i set "Use NameId for email" to True since the SAML User Unique Identifier was an email address.  i don't remember, off the top of my head, how the `missing_redirect_uri` problem was resolved.

This question remains unresolved:

 > Is SAML v2 Logout URL supported in FusionAuth's SAML configuration?

Otherwise, i don't have any further needs here.  Maybe my progress can be used by someone who might write documentation for an Azure AD SAMLv2 provider.

**Comment by mooreds** [comment]
> Is SAML v2 Logout URL supported in FusionAuth's SAML configuration?

I believe so. There is a `Logout URL` documented here: https://fusionauth.io/docs/v1/tech/samlv2/

Let me know if that answers your question and I'll close this issue (I may also, as you suggest, pull your hard won wisdom into the official docs).

**Comment by smcoll** [comment]
It does, thank you

**Comment by atrauzzi** [comment]
Just attempting to get set up with SAML and Azure Active Directory.  Have received the same error:

```json
{
  "error" : "invalid_request",
  "error_description" : "The request is missing a required parameter: redirect_uri",
  "error_reason" : "missing_redirect_uri"
}
```

**Comment by robotdan** [comment]
TL;DR

@atrauzzi I think what you're after is IdP initiated login. This is not currently supported in FusionAuth. 
https://github.com/FusionAuth/fusionauth-issues/issues/566

Longer answer - when you use a traditional SP initiated login, FusionAuth is the SP and we perform a sub-OAuth login under the SAML login, and the error you are seeing is a side effect of the fact that we did not initiate the request.

**Comment by mooreds** [comment]
@atrauzzi FYI, IdP initiated login is now available as of 1.26.0. https://fusionauth.io/docs/v1/tech/release-notes/#version-1-26-0 Full disclosure, it is a paid feature.

**Comment by david-morenomoreno** [comment]
Hey @mooreds !
Has this issue been resolved? I am getting the same error reported above on paid plan and version 1.28.1

Thanks in advance!

**Comment by mooreds** [comment]
@david-morenomoreno I don't understand your question.

Can you please provide more information and repro steps? It also might make sense to open a separate bug, as this one is about documentation. Also, if you have a plan with support, feel free to open a ticket.

**Comment by mooreds** [comment]
I'm going to close this because we have doc about Azure AD and SAML here: https://fusionauth.io/docs/lifecycle/authenticate-users/identity-providers/enterprise/azure-ad-saml

If this doc is insufficient please re-open the issue or open a new one.
